scripts/combine/setupCombine.py

```diff
@@ -767,6 +778,46 @@ def assertSample(name, startsWith=["W", "Z"], excludeMatch=[]):
                 actionArgs=dict(variations_scf=True),
             )
 
+        if args.fakeSmoothingMode == "full":
+            # add nominal nonclosure of QCD MC as single systematic
+            def fake_nonclosure(h, axesToDecorrNames, *args, **kwargs):
+                # apply QCD MC nonclosure by adding parameters (assumes log space, e.g. in full smoothing)
+                fakeselector.external_params = params_d - params
+                hvar = fakeselector.get_hist(h, *args, **kwargs)
+                # reset external parameters
+                fakeselector.external_params = None
+
+                if len(axesToDecorrNames) == 0:
+                    # inclusive
+                    hvar = hist.Hist(
+                        *hvar.axes, hist.axis.Integer(0, 1, name="var", underflow=False, overflow=False), 
+                        storage=hist.storage.Double(),
+                        data = hvar.values(flow=True)[...,np.newaxis]
+                    )
+                else:
+                    hnom = fakeselector.get_hist(h, *args, **kwargs)
+                    hvar = syst_tools.decorrelateByAxes(hvar, hnom, axesToDecorrNames)
+
+                return hvar
+
+            for axesToDecorrNames in [[], ["eta"]]:
+                subgroup = f"{cardTool.getFakeName()}QCDMCNonclosure"
+                cardTool.addSystematic(
+                    name=inputBaseName, 
+                    group="Fake",
+                    rename=subgroup+ (f"_{'_'.join(axesToDecorrNames)}" if len(axesToDecorrNames) else ""),
+                    splitGroup = {subgroup: f".*", "experiment": ".*"},
+                    systNamePrepend=subgroup,
+                    processes=cardTool.getFakeName(),
+                    noConstraint=False,
+                    mirror=True,
+                    scale=1,
+                    applySelection=False, # don't apply selection, external parameters need to be added
+                    action=fake_nonclosure,
+                    actionArgs=dict(axesToDecorrNames=axesToDecorrNames),
+                    systAxes=["var"] if len(axesToDecorrNames)==0 else [f"{n}_decorr" for n in axesToDecorrNames],
```

Comment:
does "var" still need to be in the list of systAxes for the decorrelated case?

Reply:
no